src/matter/cluster/server/GeneralCommissioningServer.ts

```diff
@@ -14,9 +18,18 @@ export const GeneralCommissioningClusterHandler: ClusterServerHandlers<typeof Ge
         return SuccessResponse;
     },
 
-    updateRegulatoryConfig: async ({request: {breadcrumbStep, config}, attributes: {breadcrumb, regulatoryConfig}}) => {
+    setRegulatoryConfig: async ({request: {breadcrumbStep, newRegulatoryConfig, countryCode}, attributes: {breadcrumb, regulatoryConfig, locationCapability}}) => {
+        const locationCapabilityValue = locationCapability.get();
+        if (locationCapabilityValue !== RegulatoryLocationType.IndoorOutdoor && newRegulatoryConfig !== locationCapabilityValue) {
```

Comment:
It is hard to follow when this is triggering.
Maybe better to make the checks explicit:
if ((locationCapabilityValue === RegulatoryLocationType.Outdoor && (newRegulatoryConfig !== RegulatoryLocationType.Outdoor))
   return {errorCode: CommissioningError.ValueOutsideRange, debugText: "Invalid regulatory location"};
if ((locationCapabilityValue === RegulatoryLocationType.Indoor && (newRegulatoryConfig !== RegulatoryLocationType.Indoor))
   return {errorCode: CommissioningError.ValueOutsideRange, debugText: "Invalid regulatory location"};
if ((locationCapabilityValue === RegulatoryLocationType.IndoorOutdoor && (newRegulatoryConfig !== RegulatoryLocationType.Indoor || newRegulatoryConfig !== RegulatoryLocationType.Outdoor))
   return {errorCode: CommissioningError.ValueOutsideRange, debugText: "Invalid regulatory location"};

Alternatively:
let validValues;
switch (locationCapabilityValue) {
  case (RegulatoryLocationType.Outdoor):
      validValues = [RegulatoryLocationType.Outdoor];
      break;
   case (RegulatoryLocationType.Indoor):
      validValues = [RegulatoryLocationType.Indoor];
      break;
   case (RegulatoryLocationType.IndoorOutdoor):
      validValues = [RegulatoryLocationType.Indoor, RegulatoryLocationType.Outdoor];
      break;
   default:
     throw new Error(`Unsupported location type: ${locationCapabilityValue}`);
}

if (validValues.findIndex(newRegulatoryConfig) === -1) 
  return {errorCode: CommissioningError.ValueOutsideRange, debugText: "Invalid regulatory location"};

Reply:
Ok, I chose the second option because in my eyes better to read then the "big if clauses". Good idea